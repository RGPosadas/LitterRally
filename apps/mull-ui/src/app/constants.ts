export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
