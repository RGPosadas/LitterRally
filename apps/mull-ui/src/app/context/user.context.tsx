import React from 'react';

const UserContext = React.createContext<{
  userId: number;
  setUserId: React.Dispatch<React.SetStateAction<number>>;
  accessToken: string;
  setAccessToken: React.Dispatch<React.SetStateAction<string>>;
}>(null);

export const UserProvider = UserContext.Provider;
export const UserConsumer = UserContext.Consumer;

export default UserContext;
