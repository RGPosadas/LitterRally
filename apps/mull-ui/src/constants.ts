import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
