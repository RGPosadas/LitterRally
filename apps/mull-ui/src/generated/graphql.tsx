import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** A date-time string at UTC, such as 2019-12-03T09:54:33Z, compliant with the date-time format. */
  DateTime: any;
  /** The `Upload` scalar type represents a file upload. */
  Upload: any;
};

export type CreateEventInput = {
  description: Scalars['String'];
  endDate: Scalars['DateTime'];
  image: MediaInput;
  restriction: Scalars['Int'];
  startDate: Scalars['DateTime'];
  title: Scalars['String'];
};

export type CreateUserInput = {
  dob?: Maybe<Scalars['DateTime']>;
  email: Scalars['String'];
  name: Scalars['String'];
  password?: Maybe<Scalars['String']>;
  registrationMethod: RegistrationMethod;
};

export type Event = {
  __typename?: 'Event';
  description: Scalars['String'];
  endDate: Scalars['DateTime'];
  id: Scalars['ID'];
  image?: Maybe<Media>;
  restriction: Scalars['Float'];
  startDate: Scalars['DateTime'];
  title: Scalars['String'];
};

export type LoginInput = {
  email: Scalars['String'];
  password: Scalars['String'];
};

export type LoginResult = {
  __typename?: 'LoginResult';
  accessToken: Scalars['String'];
};

export type Media = {
  __typename?: 'Media';
  id: Scalars['Float'];
  mediaType: Scalars['String'];
};

export type MediaInput = {
  id: Scalars['Float'];
  mediaType: Scalars['String'];
};

export type Mutation = {
  __typename?: 'Mutation';
  addParticipantToEvent: Scalars['Boolean'];
  createEvent: Event;
  createUser: User;
  deleteEvent: Event;
  deleteUser: User;
  login: LoginResult;
  removeParticipantFromEvent: Scalars['Boolean'];
  updateEvent: Event;
  updateUser: User;
  uploadFile: Media;
};

export type MutationAddParticipantToEventArgs = {
  eventId: Scalars['Float'];
  userId: Scalars['Float'];
};

export type MutationCreateEventArgs = {
  createEventInput: CreateEventInput;
};

export type MutationCreateUserArgs = {
  createUserInput: CreateUserInput;
};

export type MutationDeleteEventArgs = {
  id: Scalars['Int'];
};

export type MutationDeleteUserArgs = {
  id: Scalars['Int'];
};

export type MutationLoginArgs = {
  loginInput: LoginInput;
};

export type MutationRemoveParticipantFromEventArgs = {
  eventId: Scalars['Float'];
  userId: Scalars['Float'];
};

export type MutationUpdateEventArgs = {
  updateEventInput: UpdateEventInput;
};

export type MutationUpdateUserArgs = {
  updateUserInput: UpdateUserInput;
};

export type MutationUploadFileArgs = {
  file: Scalars['Upload'];
};

export type Query = {
  __typename?: 'Query';
  coHostEvents: Array<Event>;
  discoverEvents: Array<Event>;
  event: Event;
  events: Array<Event>;
  hostEvents: Array<Event>;
  participatingEvents: Array<Event>;
  user: User;
  users: Array<User>;
};

export type QueryCoHostEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryDiscoverEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryEventArgs = {
  id: Scalars['Int'];
};

export type QueryHostEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryParticipatingEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryUserArgs = {
  id: Scalars['Int'];
};

export enum RegistrationMethod {
  Facebook = 'FACEBOOK',
  Google = 'GOOGLE',
  Local = 'LOCAL',
  Twitter = 'TWITTER',
}

export type UpdateEventInput = {
  description?: Maybe<Scalars['String']>;
  endDate?: Maybe<Scalars['DateTime']>;
  id: Scalars['ID'];
  restriction?: Maybe<Scalars['Int']>;
  startDate?: Maybe<Scalars['DateTime']>;
  title?: Maybe<Scalars['String']>;
};

export type UpdateUserInput = {
  dob?: Maybe<Scalars['DateTime']>;
  email?: Maybe<Scalars['String']>;
  id: Scalars['ID'];
  name?: Maybe<Scalars['String']>;
};

export type User = {
  __typename?: 'User';
  description: Scalars['String'];
  dob: Scalars['DateTime'];
  email: Scalars['String'];
  friends: Array<User>;
  id: Scalars['ID'];
  name: Scalars['String'];
  password?: Maybe<Scalars['String']>;
  registrationMethod: RegistrationMethod;
  timezone: Scalars['String'];
};

export type CreateEventMutationVariables = Exact<{
  createEventInput: CreateEventInput;
}>;

export type CreateEventMutation = { __typename?: 'Mutation' } & {
  createEvent: { __typename?: 'Event' } & Pick<Event, 'id'>;
};

export type UploadFileMutationVariables = Exact<{
  file: Scalars['Upload'];
}>;

export type UploadFileMutation = { __typename?: 'Mutation' } & {
  uploadFile: { __typename?: 'Media' } & Pick<Media, 'id' | 'mediaType'>;
};

export type FindSpecificEventQueryVariables = Exact<{
  eventId: Scalars['Int'];
}>;

export type FindSpecificEventQuery = { __typename?: 'Query' } & {
  event: { __typename?: 'Event' } & Pick<
    Event,
    'id' | 'title' | 'description' | 'startDate' | 'endDate' | 'restriction'
  >;
};

export type DiscoverEventsQueryVariables = Exact<{
  discoverEventsUserId: Scalars['Int'];
}>;

export type DiscoverEventsQuery = { __typename?: 'Query' } & {
  discoverEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
};

export type GetUsersEventsQueryVariables = Exact<{
  UserId: Scalars['Int'];
}>;

export type GetUsersEventsQuery = { __typename?: 'Query' } & {
  coHostEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
  hostEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
};

export type GetParticipatingEventsQueryVariables = Exact<{
  participatingEventsUserId: Scalars['Int'];
}>;

export type GetParticipatingEventsQuery = { __typename?: 'Query' } & {
  participatingEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'endDate' | 'description' | 'startDate' | 'title'>
  >;
};

export const CreateEventDocument = gql`
  mutation CreateEvent($createEventInput: CreateEventInput!) {
    createEvent(createEventInput: $createEventInput) {
      id
    }
  }
`;
export type CreateEventMutationFn = Apollo.MutationFunction<
  CreateEventMutation,
  CreateEventMutationVariables
>;

/**
 * __useCreateEventMutation__
 *
 * To run a mutation, you first call `useCreateEventMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useCreateEventMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [createEventMutation, { data, loading, error }] = useCreateEventMutation({
 *   variables: {
 *      createEventInput: // value for 'createEventInput'
 *   },
 * });
 */
export function useCreateEventMutation(
  baseOptions?: Apollo.MutationHookOptions<CreateEventMutation, CreateEventMutationVariables>
) {
  return Apollo.useMutation<CreateEventMutation, CreateEventMutationVariables>(
    CreateEventDocument,
    baseOptions
  );
}
export type CreateEventMutationHookResult = ReturnType<typeof useCreateEventMutation>;
export type CreateEventMutationResult = Apollo.MutationResult<CreateEventMutation>;
export type CreateEventMutationOptions = Apollo.BaseMutationOptions<
  CreateEventMutation,
  CreateEventMutationVariables
>;
export const UploadFileDocument = gql`
  mutation UploadFile($file: Upload!) {
    uploadFile(file: $file) {
      id
      mediaType
    }
  }
`;
export type UploadFileMutationFn = Apollo.MutationFunction<
  UploadFileMutation,
  UploadFileMutationVariables
>;

/**
 * __useUploadFileMutation__
 *
 * To run a mutation, you first call `useUploadFileMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useUploadFileMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [uploadFileMutation, { data, loading, error }] = useUploadFileMutation({
 *   variables: {
 *      file: // value for 'file'
 *   },
 * });
 */
export function useUploadFileMutation(
  baseOptions?: Apollo.MutationHookOptions<UploadFileMutation, UploadFileMutationVariables>
) {
  return Apollo.useMutation<UploadFileMutation, UploadFileMutationVariables>(
    UploadFileDocument,
    baseOptions
  );
}
export type UploadFileMutationHookResult = ReturnType<typeof useUploadFileMutation>;
export type UploadFileMutationResult = Apollo.MutationResult<UploadFileMutation>;
export type UploadFileMutationOptions = Apollo.BaseMutationOptions<
  UploadFileMutation,
  UploadFileMutationVariables
>;
export const FindSpecificEventDocument = gql`
  query FindSpecificEvent($eventId: Int!) {
    event(id: $eventId) {
      id
      title
      description
      startDate
      endDate
      restriction
    }
  }
`;

/**
 * __useFindSpecificEventQuery__
 *
 * To run a query within a React component, call `useFindSpecificEventQuery` and pass it any options that fit your needs.
 * When your component renders, `useFindSpecificEventQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useFindSpecificEventQuery({
 *   variables: {
 *      eventId: // value for 'eventId'
 *   },
 * });
 */
export function useFindSpecificEventQuery(
  baseOptions: Apollo.QueryHookOptions<FindSpecificEventQuery, FindSpecificEventQueryVariables>
) {
  return Apollo.useQuery<FindSpecificEventQuery, FindSpecificEventQueryVariables>(
    FindSpecificEventDocument,
    baseOptions
  );
}
export function useFindSpecificEventLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<FindSpecificEventQuery, FindSpecificEventQueryVariables>
) {
  return Apollo.useLazyQuery<FindSpecificEventQuery, FindSpecificEventQueryVariables>(
    FindSpecificEventDocument,
    baseOptions
  );
}
export type FindSpecificEventQueryHookResult = ReturnType<typeof useFindSpecificEventQuery>;
export type FindSpecificEventLazyQueryHookResult = ReturnType<typeof useFindSpecificEventLazyQuery>;
export type FindSpecificEventQueryResult = Apollo.QueryResult<
  FindSpecificEventQuery,
  FindSpecificEventQueryVariables
>;
export const DiscoverEventsDocument = gql`
  query DiscoverEvents($discoverEventsUserId: Int!) {
    discoverEvents(userId: $discoverEventsUserId) {
      id
      title
      description
      startDate
      endDate
    }
  }
`;

/**
 * __useDiscoverEventsQuery__
 *
 * To run a query within a React component, call `useDiscoverEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useDiscoverEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useDiscoverEventsQuery({
 *   variables: {
 *      discoverEventsUserId: // value for 'discoverEventsUserId'
 *   },
 * });
 */
export function useDiscoverEventsQuery(
  baseOptions: Apollo.QueryHookOptions<DiscoverEventsQuery, DiscoverEventsQueryVariables>
) {
  return Apollo.useQuery<DiscoverEventsQuery, DiscoverEventsQueryVariables>(
    DiscoverEventsDocument,
    baseOptions
  );
}
export function useDiscoverEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<DiscoverEventsQuery, DiscoverEventsQueryVariables>
) {
  return Apollo.useLazyQuery<DiscoverEventsQuery, DiscoverEventsQueryVariables>(
    DiscoverEventsDocument,
    baseOptions
  );
}
export type DiscoverEventsQueryHookResult = ReturnType<typeof useDiscoverEventsQuery>;
export type DiscoverEventsLazyQueryHookResult = ReturnType<typeof useDiscoverEventsLazyQuery>;
export type DiscoverEventsQueryResult = Apollo.QueryResult<
  DiscoverEventsQuery,
  DiscoverEventsQueryVariables
>;
export const GetUsersEventsDocument = gql`
  query GetUsersEvents($UserId: Int!) {
    coHostEvents(userId: $UserId) {
      id
      title
      description
      startDate
      endDate
    }
    hostEvents(userId: $UserId) {
      id
      title
      description
      startDate
      endDate
    }
  }
`;

/**
 * __useGetUsersEventsQuery__
 *
 * To run a query within a React component, call `useGetUsersEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetUsersEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetUsersEventsQuery({
 *   variables: {
 *      UserId: // value for 'UserId'
 *   },
 * });
 */
export function useGetUsersEventsQuery(
  baseOptions: Apollo.QueryHookOptions<GetUsersEventsQuery, GetUsersEventsQueryVariables>
) {
  return Apollo.useQuery<GetUsersEventsQuery, GetUsersEventsQueryVariables>(
    GetUsersEventsDocument,
    baseOptions
  );
}
export function useGetUsersEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<GetUsersEventsQuery, GetUsersEventsQueryVariables>
) {
  return Apollo.useLazyQuery<GetUsersEventsQuery, GetUsersEventsQueryVariables>(
    GetUsersEventsDocument,
    baseOptions
  );
}
export type GetUsersEventsQueryHookResult = ReturnType<typeof useGetUsersEventsQuery>;
export type GetUsersEventsLazyQueryHookResult = ReturnType<typeof useGetUsersEventsLazyQuery>;
export type GetUsersEventsQueryResult = Apollo.QueryResult<
  GetUsersEventsQuery,
  GetUsersEventsQueryVariables
>;
export const GetParticipatingEventsDocument = gql`
  query GetParticipatingEvents($participatingEventsUserId: Int!) {
    participatingEvents(userId: $participatingEventsUserId) {
      id
      endDate
      description
      startDate
      title
    }
  }
`;

/**
 * __useGetParticipatingEventsQuery__
 *
 * To run a query within a React component, call `useGetParticipatingEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetParticipatingEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetParticipatingEventsQuery({
 *   variables: {
 *      participatingEventsUserId: // value for 'participatingEventsUserId'
 *   },
 * });
 */
export function useGetParticipatingEventsQuery(
  baseOptions: Apollo.QueryHookOptions<
    GetParticipatingEventsQuery,
    GetParticipatingEventsQueryVariables
  >
) {
  return Apollo.useQuery<GetParticipatingEventsQuery, GetParticipatingEventsQueryVariables>(
    GetParticipatingEventsDocument,
    baseOptions
  );
}
export function useGetParticipatingEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<
    GetParticipatingEventsQuery,
    GetParticipatingEventsQueryVariables
  >
) {
  return Apollo.useLazyQuery<GetParticipatingEventsQuery, GetParticipatingEventsQueryVariables>(
    GetParticipatingEventsDocument,
    baseOptions
  );
}
export type GetParticipatingEventsQueryHookResult = ReturnType<
  typeof useGetParticipatingEventsQuery
>;
export type GetParticipatingEventsLazyQueryHookResult = ReturnType<
  typeof useGetParticipatingEventsLazyQuery
>;
export type GetParticipatingEventsQueryResult = Apollo.QueryResult<
  GetParticipatingEventsQuery,
  GetParticipatingEventsQueryVariables
>;
