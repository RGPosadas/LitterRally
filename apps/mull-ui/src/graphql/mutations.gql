# Creating Events
mutation CreateEvent($event: CreateEventInput!) {
  createEvent(event: $event) {
    id
  }
}

# Uploading media files
mutation UploadFile($file: Upload!) {
  uploadFile(file: $file) {
    id
    mediaType
  }
}

# Register a user
mutation CreateUser($user: CreateUserInput!) {
  createUser(user: $user) {
    id
  }
}

# Login a user
mutation Login($loginInput: LoginInput!) {
  login(loginInput: $loginInput) {
    accessToken
  }
# User joins an event
mutation JoinEvent($eventId: Int!, $userId: Int!) {
  addParticipantToEvent(eventId: $eventId, userId: $userId)
}

# User leaves an event
mutation LeaveEvent($eventId: Int!, $userId: Int!) {
  removeParticipantFromEvent(eventId: $eventId, userId: $userId)
}
