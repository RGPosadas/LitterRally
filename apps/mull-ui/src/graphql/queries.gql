fragment EventPageContent on Event {
  id
  title
  description
  startDate
  endDate
  restriction
  location {
    title
  }
  image {
    id
    mediaType
  }
  host {
    id
    name
  }
}

fragment EventCardContent on Event {
  id
  title
  restriction
  startDate
  location {
    title
  }
  image {
    id
    mediaType
  }
}

query DiscoverEvents {
  discoverEvents {
    ...EventCardContent
  }
}

# Gets all related events to a user
query OwnedEvents {
  coHostEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
  hostEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
}

# Get all the events that a user is participating in
query ParticipantEvents {
  participantEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
}

# Find events by Id and finds participating events of user
query EventPage($eventId: Int!) {
  isParticipant(eventId: $eventId)

  event(id: $eventId) {
    ...EventPageContent
  }
}

query User {
  user {
    id
    name
    description
    avatar {
      id
    }
  }
}

query UserProfile {
  user {
    name
    description
    joinDate
    avatar {
      id
    }
  }

  friendCount
  hostingCount
  portfolioCount
}
