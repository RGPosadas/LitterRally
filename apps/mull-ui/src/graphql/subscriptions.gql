subscription PostAdded($channelId: Int!) {
  postAdded(channelId: $channelId) {
    id
    createdTime
    message
    user {
      id
      name
      avatar {
        id
        mediaType
      }
    }
  }
}
