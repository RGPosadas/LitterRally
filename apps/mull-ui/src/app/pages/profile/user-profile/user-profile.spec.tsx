import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { ROUTES } from '../../../../constants';
import { UserProfileDocument, UserProfileQuery } from '../../../../generated/graphql';
import { UserProvider } from '../../../context/user.context';
import UserProfilePage from './user-profile';

describe('UserProfilePage', () => {
  const renderHelper = (history, mocks: MockedResponse[]) => {
    return (
      <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Router history={history}>
            <UserProfilePage />
          </Router>
        </MockedProvider>
      </UserProvider>
    );
  };
  it('should render successfully', () => {
    const history = createMemoryHistory();
    history.push(ROUTES.PROFILE.DISPLAY);

    const { baseElement } = render(renderHelper(history, null));
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: UserProfileDocument,
          variables: {
            id: 1,
          },
        },
        result: {
          data: {
            user: {
              id: 1,
              name: 'Jane Doe',
              description: 'a description',
              avatar: null,
              joinDate: '2021-02-19T14:46:13.000Z',
            },
            friendCount: 0,
            hostingCount: 1,
            portfolioCount: 0,
          } as UserProfileQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.PROFILE.DISPLAY);
    await act(async () => {
      const tree = renderer.create(renderHelper(history, mocks));
      await new Promise((resolve) => setTimeout(resolve, 100));
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
