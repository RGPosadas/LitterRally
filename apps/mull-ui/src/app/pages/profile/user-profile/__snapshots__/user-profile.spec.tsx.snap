// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfilePage should match snapshot 1`] = `
<div
  className="page-container"
>
  <div
    className="profile-header-container"
  >
    <div
      className="user-name-container"
    >
      <h1
        data-testid="userName"
      >
        Jane Doe
      </h1>
    </div>
    <div
      className="profile-header"
    >
      <img
        alt="user"
        className="user-profile-picture"
        src="./assets/icons/icon-192x192.png"
      />
      <div
        className="profile-side-container for-current-user"
      >
        <button
          className="profile-stats"
          data-testid="portfolioCount"
        >
          0
          <br />
          Portfolio
        </button>
        <button
          className="profile-stats"
          data-testid="friendCount"
        >
          0
          <br />
          Friends
        </button>
        <button
          className="profile-stats"
          data-testid="hostingCount"
        >
          1
          <br />
          Hosting
        </button>
      </div>
    </div>
    <p
      className="profile-header-user-description"
      data-testid="userDescription"
    >
      a description
    </p>
  </div>
  <div
    className="settings-container"
  >
    <h3>
      Portfolio
    </h3>
    <button
      className="settings-button-container"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-leaf fa-w-18 settings-icon"
        data-icon="leaf"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M546.2 9.7c-5.6-12.5-21.6-13-28.3-1.2C486.9 62.4 431.4 96 368 96h-80C182 96 96 182 96 288c0 7 .8 13.7 1.5 20.5C161.3 262.8 253.4 224 384 224c8.8 0 16 7.2 16 16s-7.2 16-16 16C132.6 256 26 410.1 2.4 468c-6.6 16.3 1.2 34.9 17.5 41.6 16.4 6.8 35-1.1 41.8-17.3 1.5-3.6 20.9-47.9 71.9-90.6 32.4 43.9 94 85.8 174.9 77.2C465.5 467.5 576 326.7 576 154.3c0-50.2-10.8-102.2-29.8-144.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      My Portfolio
    </button>
  </div>
  <div
    className="settings-container with-friends"
  >
    <h3>
      Friends
    </h3>
    <a
      href="/profile/add-friends"
      onClick={[Function]}
    >
      <button
        className="settings-button-container"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-plus fa-w-20 settings-icon"
          data-icon="user-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Add Friends
      </button>
    </a>
    <a
      href="/my-friends"
      onClick={[Function]}
    >
      <button
        className="settings-button-container"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-friends fa-w-20 settings-icon"
          data-icon="user-friends"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        My Friends
      </button>
    </a>
    <p
      className="friend-request-count"
    >
      4
    </p>
  </div>
  <div
    className="settings-container below-friends"
  >
    <h3>
      Misc.
    </h3>
    <a
      href="/profile/edit"
      onClick={[Function]}
    >
      <button
        className="settings-button-container"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-pencil-alt fa-w-16 settings-icon"
          data-icon="pencil-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Edit Profile
      </button>
    </a>
    <button
      className="settings-button-container"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-cog fa-w-16 settings-icon"
        data-icon="cog"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Settings
    </button>
  </div>
  <div
    className="joined-date-container"
  >
    <p>
      Joined MÃ¼ll on 
      February
       
      19
      , 
      2021
    </p>
  </div>
</div>
`;
